authors/apps/articles/views.py

```diff
@@ -238,9 +244,82 @@ def delete(self, request, **kwargs):
             status.HTTP_200_OK
         )
 
+class ShareOnFaceBookView(APIView):
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (JSONRenderer, )
+
+
+    def post(self, request, *args, **kwargs):
+        server_url = f"https://{get_current_site(request)}"
+        articles_endpoint = 'api/articles'
+        facebook_share_link = f"{server_url}/{articles_endpoint}/{kwargs.get('article_id')}/"
+        facebook_share_url = "https://www.facebook.com/sharer/sharer.php?u="
+        message = {"facebook_share_link": f"{facebook_share_url}{facebook_share_link}"}
+        return Response(message)
+
+class ShareOnTwitterView(APIView):
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (JSONRenderer, )
+
+    def post(self, request, *args, **kwargs):
+        server_url = f"https://{get_current_site(request)}"
+        articles_endpoint = 'api/articles'
+        twitter_share_link = f"{server_url}/{articles_endpoint}/{kwargs.get('article_id')}/"
+        twitter_share_url = "https://twitter.com/home?status="
+        message = {"twitter_share_link": f"{twitter_share_url}{twitter_share_link}"}
+        return Response(message)
+
+class ShareEmailView(ModelViewSet):
+    """creating links for sharing articles via facebook
+    """
+
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (JSONRenderer, )
+    serializer_class = EmailSerializer
+
+    def create(self, request, *args, **kwargs):
+        article = Article.objects.get(id=self.kwargs.get("article_id"))
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        reciever_mail = serializer.data['email']
+        user_obj = User.objects.get(username=request.user.username)
+        user_serializer = UserSerializer(user_obj)
+        article_serializer = ArticleSerializer(article)
+        article_id = article.id
+        sender_email = request.user.email
```

Comment:
This block of code is quite hard to read. Please group similar function statements with a line break.